### PR TITLE
:book: Fix markdown links to flaky pages

### PR DESCRIPTION
**Pull request by killianmuldoon**
Improve the signal of the weekly markdown link checker by changing the format of flaky external links.

Fixes #

### CODE REVIEW

**Comment by killianmuldoon** [comment]
Should be cherry-picked back to supported branches.

/cherry-pick release-1.3

**Review by elmiko — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
How does the format influence the flakiness?

**Comment by sbueringer** [comment]
Makes sense

/lgtm

Not sure how to assess if the new pages are fine as well.

Stuipd question, how important / sustainable is it to have a link to Windows licensing terms in our proposal?
(cc @CecileRobertMichon @jackfrancis)

**Comment by killianmuldoon** [comment]
>Not sure how to assess if the new pages are fine as well.

I used this to check the link multiple times - ran into the same flake with the original, but nothing with the learn.microsoft.com link.

**Comment by CecileRobertMichon** [comment]
> Stupid question, how important / sustainable is it to have a link to Windows licensing terms in our proposal?
(cc @CecileRobertMichon @jackfrancis)

> Just considering how much time we already spent on those flaky links makes me wonder if it's worth the effort.

@jsturtevant @marosset @dtzar to help answer this

**Comment by marosset** [comment]
> > Stupid question, how important / sustainable is it to have a link to Windows licensing terms in our proposal?
> > (cc @CecileRobertMichon @jackfrancis)
> 
> > Just considering how much time we already spent on those flaky links makes me wonder if it's worth the effort.
> 
> @jsturtevant @marosset @dtzar to help answer this

I think it would be good to keep it.

File: docs/proposals/20200804-windows-support.md [hunk 1]
```diff
@@ -209,7 +209,7 @@ to issue [#3761](https://github.com/kubernetes-sigs/cluster-api/issues/3761) for
   - This project is under Apache 2.0 License : https://github.com/cloudbase/cloudbase-init which is cleared under the CNCF Allow list: https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md
 - Windows image Distribution
   - Infrastructure providers can provide the ability to use user provided images and images provided by image-promoter are recommended for testing and demonstration purposes. It is recommended the user creates their own image. 
-  - Users using the image scripts must ensure they are following [Windows licensing requirements](https://www.microsoft.com/en-us/licensing/product-licensing/windows-server)
+  - Users using the image scripts must ensure they are following [Windows licensing requirements](https://learn.microsoft.com/en-us/windows-server/windows-server-licensing/windows-server-licensing)
```
Comment (marosset) [inline docs/proposals/20200804-windows-support.md:212]:
```suggestion
  - Users using the image scripts must ensure they are following [Windows licensing requirements](https://learn.microsoft.com/windows-server/windows-server-licensing/windows-server-licensing)
```

**Comment by jsturtevant** [comment]
I agree it good to have it mentioned. Is there a way to disable the linter?

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**